### PR TITLE
add trim macro (#242)

### LINKED ISSUE #242: make trim macro

### PR DESCRIPTION
**Pull request by ctjhoa**
Fix #242

### CODE REVIEW

File: tests/integration/string/trim-test.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    assert,
+    computed: trim('source'),
+    properties: {
+      source: undefined
+    },
+    strictEqual: undefined
+  });
+});
+
+test('upper cases string', function(assert) {
```
Comment (mriska) [inline tests/integration/string/trim-test.js]: I assume the description here is a copy/paste error? Should probably be `trims string`.

File: tests/integration/string/trim-test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+test('returns undefined when composed undefined', function(assert) {
+  compute({
+    assert,
+    computed: trim(raw(undefined)),
+    strictEqual: undefined
+  });
+});
+
+test('upper cases composed string', function(assert) {
```
Comment (mriska) [inline tests/integration/string/trim-test.js]: I assume the description here is a copy/paste error? Should probably be `trims composed string`.

File: addon/index.js [hunk 3]
```diff
@@ -87,6 +87,7 @@ const substr = deprecate(string.substr, 'substr', 'string.substr');
 const substring = deprecate(string.substring, 'substring', 'string.substring');
 const toLower = deprecate(string.toLower, 'toLower', 'string.toLower');
 const toUpper = deprecate(string.toUpper, 'toUpper', 'string.toUpper');
+const trim = deprecate(string.trim, 'trim', 'string.trim');
```
Comment (mriska) [inline addon/index.js]: Is this correct? Has `trim` existed previously and been deprecated?

File: tests/integration/string/trim-test.js (hunk 1 above)
Comment (ctjhoa) [reply tests/integration/string/trim-test.js]: You're right just basd copy paste.

File: tests/integration/string/trim-test.js (hunk 2 above)
Comment (ctjhoa) [reply tests/integration/string/trim-test.js]: You're right just basd copy paste.

File: addon/index.js (hunk 3 above)
Comment (ctjhoa) [reply addon/index.js]: I have follow `toUpper` behavior. I think it should be better to remove all deprecated stuff `string.*` in one time in order to keep consistency in code.

**Comment by kellyselden** [comment]
released in 0.41.0

**Pull request merged**